petastorm/spark/spark_dataset_converter.py

```diff
@@ -199,14 +236,34 @@ class _torch_dataset_context_manager(object):
     :class:`petastorm.Reader`.
     """
 
-    def __init__(self, data_url):
+    def __init__(self, data_url, batch_size, num_epochs, workers_count,
+                 cur_shard, shard_count, preprocess_pandas_fn,
+                 **petastorm_reader_kwargs):
         """
         :param data_url: A string specifying the data URL.
+        See `SparkDatasetConverter.make_torch_dataloader()` for the definitions
+        of the other parameters.
         """
         from petastorm.pytorch import DataLoader
-
-        self.reader = make_batch_reader(data_url)
-        self.loader = DataLoader(reader=self.reader)
+        petastorm_reader_kwargs["num_epochs"] = num_epochs
+        if workers_count is not None:
+            petastorm_reader_kwargs["workers_count"] = workers_count
+        petastorm_reader_kwargs["cur_shard"] = cur_shard
+        petastorm_reader_kwargs["shard_count"] = shard_count
+        if preprocess_pandas_fn is None:
+            # If transform_spec is provided, we respect it.
+            pass
+        elif "transform_spec" not in petastorm_reader_kwargs:
+            # If transform_spec is not provided, we use preprocess_pandas_fn.
+            transform = TransformSpec(preprocess_pandas_fn, removed_fields=[])
+            petastorm_reader_kwargs["transform_spec"] = transform
+        else:
+            # If both are provided, we respect transform_spec with a warning.
+            warnings.warn("The param `preprocess_pandas_fn` will be ignored "
+                          "when `transform_spec` is provided.")
+
+        self.reader = make_batch_reader(data_url, **petastorm_reader_kwargs)
```

Comment:
We need to check whether `petastorm_reader_kwargs` dict include some params we don't allow user to set here, include:
```
data_url
transform_spec: this is mutually exclusive with preprocess_pandas_fn, you need check.
```

Reply:
Currently, if `dataset_url` is passed in as a keyword argument, petastorm should throw an error because there is no such keyword param. The check for `transform_spec` is in L253-263. We can discuss the desired behavior when `preprocess_pandas_fn` and `transform_spec` are both provided. I prefer respecting the `transform_spec` while showing a warning, since `transform_spec` matches the exact name of the upstream param.